.github/workflows/lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

Comment:
I think we can just run this on all pull requests, when I have some complicated changes I sometimes like to use [Stacked PRs](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/), this workflow would not run for them.

This also applies to `.github/workflows/test.yml`

Reply:
Actuall, it was fine having it only run on pushes to `main`. If we have `push` & `pull_request` for all branches, pull requests will have double jobs running. This is already happening on some of our projects like https://github.com/hetznercloud/hcloud-go/pull/277/checks

Reply:
Alright, fixed it.